### PR TITLE
Refactor monitoring rules

### PR DESCRIPTION
**Pull request by machadovilaca**
**What this PR does / why we need it**:

Following the work started in https://github.com/kubevirt/kubevirt/pull/10700, and according to the kubevirt/community#219 proposal, this PR refactors monitoring rules

jira-ticket: https://issues.redhat.com/browse/CNV-34289

**Special notes for your reviewer**:

**Release note**:

```release-note
Refactor monitoring rules
```

### CODE REVIEW

File: tools/prom-metrics-collector/metrics_json_generator.go [hunk 1]
```diff
@@ -40,13 +39,8 @@ func main() {
...
 	}
 
-	metricsList := metrics.ListMetrics()
-	for _, metric := range metricsparser.ReadFromPrometheusCR() {
-		metricsList = append(metricsList, metric)
-	}
-
 	var metricFamilies []parser.Metric
-	for _, m := range metricsList {
+	for _, m := range metrics.ListMetrics() {
```
Comment (avlitman) [inline tools/prom-metrics-collector/metrics_json_generator.go:43]: where is the recording rules? 
In hco Shirly asked to add them as well to the doc and linter (if the name is sum: or cluster: just ignore them in the list above), see: https://github.com/kubevirt/hostpath-provisioner-operator/pull/417/files#diff-67148ace3c0d839a4031d5be8a050b10553fcb7befa7f5b3a191d9b95ffda424R59

File: tools/prom-rule-ci/verify-rules.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
+package main
```
Comment (avlitman) [inline tools/prom-rule-ci/verify-rules.go:1]: Is this file should be added for all operators?

File: pkg/monitoring/rules/alerts/alerts.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package alerts
```
Comment (avlitman) [inline pkg/monitoring/rules/alerts/alerts.go:1]: @sradco mentioned this file can be named prometheus.go

File: tools/metricsdocs/metricsdocs.go [hunk 4]
```diff
@@ -3,10 +3,11 @@ package main
 import (
```
Comment (avlitman) [inline tools/metricsdocs/metricsdocs.go:3]: this file should move under pkg/monitoring/tools

File: tools/prom-metrics-collector/metrics_json_generator.go [hunk 5]
```diff
@@ -26,7 +26,6 @@ import (
 	"github.com/kubevirt/monitoring/pkg/metrics/parser"
```
Comment (avlitman) [inline tools/prom-metrics-collector/metrics_json_generator.go:26]: this file should move under pkg/monitoring/tools

File: go.mod [hunk 6]
```diff
@@ -23,14 +23,14 @@ require (
 	github.com/openshift/origin v4.1.0+incompatible
 	github.com/operator-framework/operator-sdk v1.12.0
 	github.com/pkg/errors v0.9.1
+	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.64.1
```
Comment (RamLavi) [inline go.mod:26]: did you run make vendor afterwards, and add the necessary files?

**Review by RamLavi — CHANGES_REQUESTED** [review]
Hey @machadovilaca - This PR lacks commit descriptions on all the commits. Please add these so I can review.

File: tools/metricsdocs/metricsdocs.go (hunk 4 above)
Comment (machadovilaca) [reply tools/metricsdocs/metricsdocs.go:3]: Tools should be in the pkg/monitoring! Many component's Dockefiles when building the operator copy pkg/ into the final image, and it is not necessary for the final build, so it should be in a separate base directory like just tools/

File: tools/prom-metrics-collector/metrics_json_generator.go (hunk 1 above)
Comment (machadovilaca) [reply tools/prom-metrics-collector/metrics_json_generator.go:49]: updated

File: tools/prom-rule-ci/verify-rules.go (hunk 2 above)
Comment (machadovilaca) [reply tools/prom-rule-ci/verify-rules.go:1]: here they already had the rules unit tests, on kubevirt they already had something similar, so it really depends

File: pkg/monitoring/rules/alerts/alerts.go (hunk 3 above)
Comment (machadovilaca) [reply pkg/monitoring/rules/alerts/alerts.go]: done

File: go.mod (hunk 6 above)
Comment (machadovilaca) [reply go.mod:26]: yes, no additional changes

File: tools/metricsdocs/metricsdocs.go (hunk 4 above)
Comment (avlitman) [reply tools/metricsdocs/metricsdocs.go:3]: In cdi we already moved the metrics tools under monitoring, and also I was able to move them in hco and hpp.
Basically it was Shirly's request so adding here @sradco

**Review by RamLavi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**